viper_test.go

```diff
@@ -2258,6 +2258,37 @@ func TestKeyDelimiter(t *testing.T) {
 	assert.Equal(t, expected, actual)
 }
 
+var yamlExampleWithNilValues = []byte(`Hacker: true
+name: steve
+hobbies:
+clothing:
+  jacket:
+`)
+
+func TestNilValues(t *testing.T) {
```

Comment:
Can you please extend this test with (unset) environment variables bound to a custom key?

Like this: `BindEnv("age", "AGE")`

Just to make sure, that env vars don't mess up anything (and `AllKeys` really returns keys with explicitly set values)